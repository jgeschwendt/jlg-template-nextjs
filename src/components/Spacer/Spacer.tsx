import React from 'react';
import styled from 'styled-components';

const Component = styled.div``;

export const Spacer = ({ children }: React.PropsWithChildren<Record<string, unknown>>): JSX.Element => {
  return (
    <Component>{children}</Component>
  );
};
