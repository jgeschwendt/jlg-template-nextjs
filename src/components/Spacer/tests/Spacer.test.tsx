import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import { Spacer } from '../Spacer';

describe('<Spacer />', () => {
  it('should be defined', () => {
    expect(Spacer).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(create(<Spacer>Text</Spacer>).toJSON()).toMatchSnapshot();
  });

  it('should work with @testing-library/react', () => {
    const text = 'Text';
    const {
      container,
    } = render(
      <Spacer>{text}</Spacer>
    );

    expect(container.className).toBeDefined();
    expect(container.querySelector('div').textContent).toEqual(text);
  });
});
