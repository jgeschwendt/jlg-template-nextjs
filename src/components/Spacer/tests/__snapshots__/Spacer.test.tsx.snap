// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> should match the snapshot 1`] = `
<div
  className="Spacer__Component-sc-2y4bj-0"
>
  Text
</div>
`;
